public/pages/utils/anomalyResultUtils.ts

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import { SORT_DIRECTION, AD_DOC_FIELDS } from '../../../server/utils/constants';
+import { getDetectorResults } from '../../redux/reducers/anomalyResults';
+import { getDetectorLiveResults } from '../../redux/reducers/liveAnomalyResults';
+import moment from 'moment';
+import { Dispatch } from 'redux';
+import { get } from 'lodash';
+import { AnomalyData, DateRange } from 'public/models/interfaces';
+
+export const getLiveAnomalyResults = (
+  dispatch: Dispatch<any>,
+  detectorId: string,
+  detectionInterval: number,
+  intervals: number
+) => {
+  const startTime = moment()
+    .subtract(intervals * detectionInterval, 'minutes')
+    .valueOf();
+  const updatedParams = {
+    from: 0,
+    size: intervals,
+    sortDirection: SORT_DIRECTION.DESC,
+    sortField: AD_DOC_FIELDS.DATA_START_TIME,
+    dateRangeFilter: {
+      startTime: startTime.valueOf(),
+      fieldName: AD_DOC_FIELDS.DATA_START_TIME,
+    },
+  };
+  dispatch(getDetectorLiveResults(detectorId, updatedParams));
+};
+
+export const getAnomalyResultsWithDateRange = (
+  dispatch: Dispatch<any>,
+  startTime: number,
+  endTime: number,
+  detectorId: string
+) => {
+  const updatedParams = {
+    from: 0,
+    size: 10000,
+    sortDirection: SORT_DIRECTION.DESC,
+    sortField: AD_DOC_FIELDS.DATA_START_TIME,
+    dateRangeFilter: {
+      startTime: startTime,
+      endTime: endTime,
+      fieldName: AD_DOC_FIELDS.DATA_START_TIME,
+    },
+  };
+  dispatch(getDetectorResults(detectorId, updatedParams));
+};
+
+const MAX_DATA_POINTS = 1000;
+
+const calculateStep = (total: number): number => {
+  return Math.ceil(total / MAX_DATA_POINTS);
+};
+
+function findAnomalyWithMaxAnomalyGrade(anomalies: any[]) {
+  let anomalyWithMaxGrade = anomalies[0];
+  for (let i = 1, len = anomalies.length; i < len; i++) {
+    let anomaly = anomalies[i];
+    anomalyWithMaxGrade =
+      anomaly.anomalyGrade > anomalyWithMaxGrade.anomalyGrade
+        ? anomaly
+        : anomalyWithMaxGrade;
+  }
+  return anomalyWithMaxGrade;
```

Comment:
may simplify by sorting it first then pick the biggest one.

Reply:
Add TODO here, will create an issue to track this too.

Reply:
https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/issues/63